pytorch_lightning/trainer/trainer.py

```diff
@@ -449,7 +449,7 @@ def __init__(
         self.tuner = Tuner(self)
 
         min_steps, max_steps, min_epochs, max_epochs, max_time = _parse_loop_limits(
-            min_steps, max_steps, min_epochs, max_epochs, max_time
+            min_steps, max_steps, min_epochs, max_epochs, max_time, fast_dev_run
```

Comment:
IMO instead of passing `fast_dev_run` here, a better solution would be to call this

https://github.com/PyTorchLightning/pytorch-lightning/blob/bfa8b7be2d99b980afa62f5cb0433326bcfd2ef0/pytorch_lightning/trainer/trainer.py#L544

before `parse_loop_limits` since `fast_dev_run` might modify the limits.

Reply:
It looks like moving `_init_debugging_flags` above `_parse_loop_limits` breaks because it requires `fit_loop` to be set, but that is set using values from the output of `_parse_loop_limits`. Should I revert to my previous solution or is there a way to break this loop?

Reply:
I pushed a commit that should fix the problem. It required inlining the `init_debugging_flags` method.